src/fabric/src/fabric2_fdb.erl

```diff
@@ -1364,3 +1371,21 @@ run_on_commit_fun(Tx) ->
             Fun(),
             ok
     end.
+
+start_span_fun(Fun, Options, ExtraTags) ->
+    {name, OperationName} = erlang:fun_info(Fun, name),
+    UserCtx = fabric2_util:get_value(user_ctx, Options, #user_ctx{}),
+    Tags = maps:merge(#{
+        'span.kind' => <<"client">>,
+        component => <<"couchdb.fabric">>,
+        'db.instance' => fabric2_server:fdb_cluster(),
+        'db.namespace' => fabric2_server:fdb_directory(),
+        'db.type' => <<"fdb">>,
+        user => UserCtx#user_ctx.name,
+        nonce => get(nonce),
+        pid => self()
+    }, ExtraTags),
+    fun() ->
+        ctrace:has_span()
```

Comment:
If we wrap `with_span` here we can get rid of passing these closures around. Regardless, it'll be slightly more performant to move all of the OpName/UserCtx/Tags creation inside the check for `ctrace:has_span()` so that when tracing is turned off we're not needlessly creating these extra terms on the heap.